### PR TITLE
fixed error: externally-managed-environment arising in pip version > 23 (ubuntu 23.04)

### PR DESCRIPTION
**Pull request by anandhu-eng**

### CODE REVIEW

File: cm-mlops/script/get-generic-python-lib/customize.py [hunk 1]
```diff
@@ -9,7 +9,8 @@ def preprocess(i):
     meta = i['meta']
     automation = i['automation']
     run_script_input = i['run_script_input']
-
+    pip_version = env.get('CM_PIP_VERSION')
```
Comment (arjunsuresh) [inline cm-mlops/script/get-generic-python-lib/customize.py:12]: get can return None too. So, we should use second argument as '' before using any string function.

File: cm-mlops/script/get-generic-python-lib/customize.py [hunk 2]
```diff
@@ -9,7 +9,8 @@ def preprocess(i):
...
+    print("PIP Version:"+str(pip_version))
```
Comment (arjunsuresh) [inline cm-mlops/script/get-generic-python-lib/customize.py:13]: Not needed to print right?

File: cm-mlops/script/get-generic-python-lib/customize.py [hunk 3]
```diff
@@ -38,8 +39,8 @@ def preprocess(i):
                 version = env.get('CM_HOST_OS_VERSION', '')
                 if version:
                     version_split = version.split(".")
-                    if (int(version_split[0]) >= 23) and ('--break-system-packages' not in extra):
-                        extra += '  --break-system-packages '
+                    if (pip_version >= "23.0.0") and ('--break-system-packages' not in extra):
```
Comment (arjunsuresh) [inline cm-mlops/script/get-generic-python-lib/customize.py:42]: String comparison using `>=`? This comparison should be done similar to the earlier one used for os version check with 3 parts.

**Review by arjunsuresh — CHANGES_REQUESTED** [review]
Thank you for the PR. Please do these changes

File: cm-mlops/script/get-generic-python-lib/customize.py [hunk 4]
```diff
@@ -38,8 +38,8 @@ def preprocess(i):
                 version = env.get('CM_HOST_OS_VERSION', '')
                 if version:
                     version_split = version.split(".")
-                    if (int(version_split[0]) >= 23) and ('--break-system-packages' not in extra):
-                        extra += '  --break-system-packages '
+                    if (int(pip_version[0]) >= 23) and ('--break-system-packages' not in extra):
```
Comment (arjunsuresh) [inline cm-mlops/script/get-generic-python-lib/customize.py:41]: now do we need the version information of ubuntu? We can remove that part right?

Comment (anandhu-eng) [reply cm-mlops/script/get-generic-python-lib/customize.py:41]: if r['return'] == 16:
            extra = env.get('CM_GENERIC_PYTHON_PIP_EXTRA','')

should we need this?

Comment (arjunsuresh) [reply cm-mlops/script/get-generic-python-lib/customize.py:41]: yes, we need that

**Review by arjunsuresh — APPROVED** [review]
LGTM

File: cm-mlops/script/get-generic-python-lib/customize.py (hunk 4 above)
Comment (anandhu-eng) [reply cm-mlops/script/get-generic-python-lib/customize.py]: I have not removed the code as "extra" variable was used later in the code

**Pull request merged**